src/phasorpy/components.py

```diff
@@ -91,26 +98,143 @@ def two_fractions_from_phasor(
         raise ValueError(f'{real_components.shape=} != (2,)')
     if imag_components.shape != (2,):
         raise ValueError(f'{imag_components.shape=} != (2,)')
-    first_component_phasor = numpy.array(
-        [real_components[0], imag_components[0]]
-    )
-    second_component_phasor = numpy.array(
-        [real_components[1], imag_components[1]]
+    _, distance_between_components = line_from_components(
+        real_components, imag_components
     )
-    total_distance_between_components = math.hypot(
-        (second_component_phasor[0] - first_component_phasor[0]),
-        (second_component_phasor[1] - first_component_phasor[1]),
-    )
-    if math.isclose(total_distance_between_components, 0, abs_tol=1e-6):
-        raise ValueError('components must have different coordinates')
     projected_real, projected_imag = project_phasor_to_line(
         real, imag, real_components, imag_components
     )
     distances_to_first_component = numpy.hypot(
-        numpy.asarray(projected_real) - first_component_phasor[0],
-        numpy.asarray(projected_imag) - first_component_phasor[1],
+        numpy.asarray(projected_real) - real_components[0],
+        numpy.asarray(projected_imag) - imag_components[0],
     )
-    fraction_of_second_component = (
-        distances_to_first_component / total_distance_between_components
+    second_component_fractions = (
+        distances_to_first_component / distance_between_components
     )
-    return 1 - fraction_of_second_component, fraction_of_second_component
+    first_component_fractions = 1 - second_component_fractions
+    return first_component_fractions, second_component_fractions
+
+
+def graphical_component_analysis(
+    real: ArrayLike,
+    imag: ArrayLike,
+    real_components: ArrayLike,
+    imag_components: ArrayLike,
+    /,
+    *,
+    cursor_diameter: float = 0.1,
+    number_of_steps: int = 100,
+) -> tuple[tuple[NDArray[Any], ...], NDArray[Any]]:
+    """Return fractions of two or three components from phasor coordinates by
+    solving graphically.
+
+    Parameters
+    ----------
+    real : array_like
+        Real component of phasor coordinates.
+    imag : array_like
+        Imaginary component of phasor coordinates.
+    real_components: array_like
```

Comment:
Yes it is a good idea to keep the format. Do you think is better in this case `components_real` instead of `component_real` since it has more than one component?